src/mri/cli/dc_adjoint.py

```diff
@@ -0,0 +1,105 @@
+from hydra_zen import store, builds, zen
+
+from mrinufft.io import read_trajectory
+from mri.operators import NonCartesianFFT
+from mrinufft.io.nsp import read_siemens_rawdat
+from mrinufft.io.utils import add_phase_to_kspace_with_shifts
+
+from mrinufft.trajectories.utils import DEFAULT_RASTER_TIME
+
+import numpy as np
+import pickle as pkl
+import logging
+import os
+
+log = logging.getLogger(__name__)
+raw_config = builds(read_siemens_rawdat, populate_full_signature=True, zen_partial=True)
+traj_config = builds(
+    read_trajectory,
+    populate_full_signature=True,
+    zen_exclude=["dwell_time"],
+    zen_partial=True,
+)
+
+fourier_op_config = builds(
+    NonCartesianFFT,
+    populate_full_signature=True,
+    zen_partial=True,
+    zen_exclude=["n_coils"],
+)
+fourier_store = store(group="fourier")
+fourier_store(fourier_op_config, name="cpu")
+fourier_store(
+    fourier_op_config, implementation="gpuNUFFT", density_comp="pipe", name="gpu"
+)
+
+
+def recon(obs_file: str, traj_file: str, obs_reader, traj_reader, fourier):
+    """
+    Reconstructs an image using the adjoint operator.
+
+    Parameters
+    ----------
+    obs_file : str
+        Path to the observed kspace data file.
+    traj_file : str
+        Path to the trajectory file.
+    obs_reader : callable
+        A function that reads the observed data file and returns
+        the raw data and data header.
+    traj_reader : callable
+        A function that reads the trajectory file and returns the trajectory
+        data and parameters.
+    fourier : callable
+        A function that performs Fourier transform operations.
```

Comment:
```suggestion
fourier: Callable
   A Callable returning a Fourier Operator
``` 
Also described what input argument it should have.